working/augmentation-libraries/feature-specification.md

```diff
@@ -616,6 +626,59 @@ It's a **compile-time** error if:
     other top-level variables that might be in scope in the library
     augmentation.*
 
+### Augmenting function and constructor signatures
+
+[signature matching]: #Augmenting_function_and_constructor_signatures
+
+When augmenting a function (top level, static method, instance method, etc.) or
+constructor (generative, factory, etc.) the parameter lists must be the same in
+all meaningful ways. We say that an augmenting function or constructor's
+signature *matches* if:
+
+*   It has the same number of type parameters with the same type parameter names
+    (same identifiers) and bounds (after type annotation inheritance), if any
+    (same *types*, even if they may not be written exactly the same in case one
+    of the declarations needs to refer to a type using an import prefix).
+
+*   The return type (after type annotation inheritance) is the same as the
+    augmented declaration's return type.
+
+*   It has the same number of positional and optional parameters as the
+    augmented declaration.
+
+*   It has the same set of named parameter names as the augmented declaration.
+
+*   For all corresponding pairs of parameters:
+
+    *   They have the same type (after type annotation inheritance).
+
+    *   They have the same `required` and `covariant` modifiers.
+
+    *   If augmenting a constructor, they must both be initializing formals,
+        super parameters, or neither.
+
+*   For all positional parameters:
+
+    *   The augmenting function's parameter name is `_`, or
+
+    *   The augmenting function's parameter name is the same as the name of the
+        corresponding positional parameter in every preceding declaration that
+        doesn't have `_` as its name.
+
+    *In other words, a declaration can ignore a positional parameter's name by
+    using `_`, but all declarations in the chain that specify a name have to
+    agree on it.*
```

Comment:
(Can an implementing augmentation omit a parameter name, and if it does so, can it refer to the parameter?
```dart
void foo(int x);
augment void foo(int _) {
  print(x); // Works or not?
}
```
(I'd say "works not!', the name of the implementing declaration is the one that matters to the implementation.)

Reply:
> Can an implementing augmentation omit a parameter name

Yes.

> and if it does so, can it refer to the parameter?

No! Added a note clarifying.